packaging/metadata.py

```diff
@@ -0,0 +1,191 @@
+
+    All values are lower-cased for easy comparison.
+    """
+
+    # `Name`, `Version`, and `Metadata-Version` are invalid in `Dynamic`.
+    # 1.0
+    PLATFORM = "platform"
+    SUMMARY = "summary"
+    DESCRIPTION = "description"
+    KEYWORDS = "keywords"
+    HOME_PAGE = "home-page"
+    AUTHOR = "author"
+    AUTHOR_EMAIL = "author-email"
+    LICENSE = "license"
+    # 1.1
+    SUPPORTED_PLATFORM = "supported-platform"
+    DOWNLOAD_URL = "download-url"
+    CLASSIFIER = "classifier"
+    # 1.2
+    MAINTAINER = "maintainer"
+    MAINTAINER_EMAIL = "maintainer-email"
+    REQUIRES_DIST = "requires-dist"
+    REQUIRES_PYTHON = "requires-python"
+    REQUIRES_EXTERNAL = "requires-external"
+    PROJECT_URL = "project-url"
+    PROVIDES_DIST = "provides-dist"
+    OBSOLETES_DIST = "obsoletes-dist"
+    # 2.1
+    DESCRIPTION_CONTENT_TYPE = "description-content-type"
+    PROVIDES_EXTRA = "provides-extra"
+
+
+class Metadata:
+
+    """
+    A representation of core metadata.
+    """
+
+    # A property named `display_name` exposes the value.
+    _display_name: str
+    # A property named `canonical_name` exposes the value.
+    _canonical_name: utils.NormalizedName
+    version: packaging_version.Version
+    platforms: Set[str]
+    summary: str
+    description: str
+    keywords: List[str]
+    home_page: str
+    author: str
+    author_emails: List[_NameAndEmail]
+    license: str
+    supported_platforms: Set[str]
+    download_url: str
+    classifiers: Set[str]
+    maintainer: str
+    maintainer_emails: List[_NameAndEmail]
+    requires_dists: Set[requirements.Requirement]
+    requires_python: specifiers.SpecifierSet
+    requires_externals: Set[str]
+    project_urls: Set[_LabelAndURL]
+    provides_dists: Set[requirements.Requirement]
+    obsoletes_dists: Set[requirements.Requirement]
+    description_content_type: str
+    provides_extras: Set[utils.NormalizedName]
+    dynamic: Set[DynamicField]
+
+    def __init__(
+        self,
+        name: str,
+        version: packaging_version.Version,
+        *,
+        # 1.0
+        platforms: Optional[Iterable[str]] = None,
+        summary: Optional[str] = None,
+        description: Optional[str] = None,
+        keywords: Optional[Iterable[str]] = None,
+        home_page: Optional[str] = None,
+        author: Optional[str] = None,
+        author_emails: Optional[Iterable[_NameAndEmail]] = None,
+        license: Optional[str] = None,
+        # 1.1
+        supported_platforms: Optional[Iterable[str]] = None,
+        download_url: Optional[str] = None,
+        classifiers: Optional[Iterable[str]] = None,  # TODO: OK?
+        # 1.2
+        maintainer: Optional[str] = None,
+        maintainer_emails: Optional[Iterable[_NameAndEmail]] = None,
+        requires_dists: Optional[Iterable[requirements.Requirement]] = None,
+        requires_python: Optional[specifiers.SpecifierSet] = None,
+        requires_externals: Optional[Iterable[str]] = None,  # TODO: OK?
```

Comment:
The only two cases of requires_external in the wheels on PyPI are "msbuild" and "rust", so `Optional[Iterable[str]]` looks fine.